src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -302,14 +302,15 @@ public void closeAll() {
   /**
    * This function is called during rebalance.
    *
-   * <p>All the channels are closed. The client is still active. Upon rebalance, (inside {@link
-   * com.snowflake.kafka.connector.SnowflakeSinkTask#open(Collection)} we will reopen the channel.
+   * <p>We don't close the channels. Upon rebalance, (inside {@link
```

Comment:
I'm a bit concern about this change, did the customer test this and verify that this actually fixes their issue? Or they just reduced the number of topics and the issue is gone. If the theory is that multiple channel open compete with each other then this change won't help, right?

Two area of concerns I have:
- This means that we're not verifying that everything is committed before rebalance and if there is a reported customer issue, we won't know that the issue is because of channel invalidation by reading through the KC log
- Resources are not released and keeps increasing, we have a cache per client and the flush service will iterate over all active channels to see if there is anything to flush, the number of active channels will just keep increasing

Reply:
These are some valid concerns and I will come back with more findings and whether we "really" need this change. Thanks Toby!